caffe2/operators/mean_op.h

```diff
@@ -20,35 +21,45 @@ class MeanOp final : public Operator<Context> {
   template <typename T>
   bool DoRunWithType() {
     auto& input0 = Input(0);
-
     auto* output = Output(0, input0.sizes(), at::dtype<T>());
     output->CopyFrom(input0, true /*async*/);
-
     if (InputSize() == 1) {
       return true;
     }
 
-    // Dimension checking
-    for (int i = 1; i < InputSize(); ++i) {
-      if (output->sizes() != Input(i).sizes()) {
-        CAFFE_THROW(
-            "Check failed: output->sizes() == Input(i).sizes().",
-            "Description: Input #",
-            i,
-            ", input dimension:",
-            Input(i).sizes(),
-            " should match output dimension: ",
-            output->sizes());
-      }
+    std::vector<std::vector<int>> input_dims(InputSize());
+    for (int i = 0; i < InputSize(); i++) {
+      std::copy(
+          Input(i).sizes().cbegin(),
+          Input(i).sizes().cend(),
+          std::back_inserter(input_dims[i]));
     }
 
-    T* output_data = output->template mutable_data<T>();
-    for (int i = 1; i < InputSize(); ++i) {
+    auto output_dims = ComputeBroadcastDims(input_dims);
+    if (IsInputOutputAlias(0, 0)) {
+      CAFFE_ENFORCE(
+          (output_dims.size() == input_dims[0].size()) &&
+              std::equal(
+                  output_dims.begin(),
+                  output_dims.begin() + output_dims.size(),
+                  input_dims[0].begin()),
+          "Cannot broadcast output to the first dimension.");
```

Comment:
This error information does not match your check.

Reply:
Would something like this be better? "Cannot broadcast output to the dimensions of the first input".